### PR TITLE
Fix query hang when a group has no splits in grouped execution

### PR DESCRIPTION
**Pull request by haozhun**

### CODE REVIEW

**Comment by wenleix** [comment]
Per offline discussion, the second commit doesn't seem necessary for this commit anymore.

File: presto-main/src/main/java/com/facebook/presto/execution/scheduler/FixedSourcePartitionedScheduler.java [hunk 1]
```diff
@@ -177,12 +178,16 @@ public ScheduleResult schedule()
...
         Iterator<SourceScheduler> schedulerIterator = sourceSchedulers.iterator();
         List<Lifespan> driverGroupsToStart = ImmutableList.of();
+        boolean shouldInvokeNoMoreDriverGroups = false;
         while (schedulerIterator.hasNext()) {
             SourceScheduler sourceScheduler = schedulerIterator.next();
 
             for (Lifespan lifespan : driverGroupsToStart) {
                 sourceScheduler.startLifespan(lifespan, partitionHandleFor(lifespan));
             }
+            if (shouldInvokeNoMoreDriverGroups) {
```
[the remark this answers is not in the document]
Comment (haozhun) [reply presto-main/src/main/java/com/facebook/presto/execution/scheduler/FixedSourcePartitionedScheduler.java]: S2 is not considered finished unless `noMoreLifespans()` has been invoked on it.

File: presto-main/src/main/java/com/facebook/presto/execution/scheduler/FixedSourcePartitionedScheduler.java [hunk 2]
```diff
@@ -300,10 +315,21 @@ public void scheduleInitial(SourceScheduler scheduler)
...
+                IntListIterator driverGroupsIterator = entry.getValue();
+                if (!driverGroupsIterator.hasNext()) {
+                    entryIterator.remove();
+                    continue;
+                }
+                int driverGroupId = driverGroupsIterator.nextInt();
                 scheduler.startLifespan(Lifespan.driverGroup(driverGroupId), partitionHandles.get(driverGroupId));
             }
+
+            if (nodeToDriverGroupsMap.isEmpty()) {
```
[the remark this answers is not in the document]
Comment (haozhun) [reply presto-main/src/main/java/com/facebook/presto/execution/scheduler/FixedSourcePartitionedScheduler.java]: I'm OK either way. I can add a `nodesWithAllDriverGroupsScheduled` set. That does seem to be simpler.

**Comment by haozhun** [comment]
Addressed coments

File: presto-main/src/main/java/com/facebook/presto/execution/scheduler/FixedSourcePartitionedScheduler.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (haozhun) [reply presto-main/src/main/java/com/facebook/presto/execution/scheduler/FixedSourcePartitionedScheduler.java]: Added `totalDriverGroupsScheduled`

**Pull request merged**